angular/src/lib/data-table/data-table.component.ts

```diff
@@ -0,0 +1,530 @@
+    // display none start
+    this.resizeHelperViewChild.nativeElement.style.display = 'block';
+  }
+
+
+  onColumnResizeEnd(event, column) {
+
+    let movedDistance = this.resizeHelperViewChild.nativeElement.offsetLeft - this.resizerX;
+    const columnWidth = column.offsetWidth;
+    const minWidth = parseInt(column.style.minWidth || 15, 10);
+
+    if (columnWidth + movedDistance < minWidth) {
+      movedDistance = minWidth - columnWidth;
+    }
+
+    const newColumnWidth = columnWidth + movedDistance;
+
+    if (newColumnWidth >= minWidth) {
+
+      let nextColumn = column.nextElementSibling;
+
+      while (!nextColumn.offsetParent) {
+        nextColumn = nextColumn.nextElementSibling;
+      }
+
+      if (nextColumn) {
+          const nextColumnWidth = nextColumn.offsetWidth - movedDistance;
+          const nextColumnMinWidth = nextColumn.style.minWidth || 15;
+
+        if (newColumnWidth > 15 && nextColumnWidth > parseInt(nextColumnMinWidth, 10)) {
+
+          column.style.width = newColumnWidth + 'px';
+
+          if (nextColumn) {
+            nextColumn.style.width = nextColumnWidth + 'px';
+          }
+        }
+      }
+
+    this.afterColResize.emit({
+      element: column,
+      movedDistance: movedDistance
+    });
+  }
+  // hide resize helper line on end
+  this.resizeHelperViewChild.nativeElement.style.display = 'none';
+  }
+}
+
+
+// Body Component
+
+@Component({
+  selector: '[bodyColumns]',
+  template: `
+    <ng-container>
+
+      <ng-template
+        ngFor let-row let-rowIndex="index"
+        [ngForOf]="dt.data"
+        [ngForTrackBy]="dt.rowTrackBy"
+      >
+
+        <ng-container
+          *ngTemplateOutlet="bodyTemplate;
+          context: {
+            $implicit: row,
+            columns: columns,
+            rowIndex: rowIndex
+          }">
+          </ng-container>
+
+      </ng-template>
+
+    </ng-container>
+
+
+    <!-- Message if empty TODO: remove if not needed-->
+
+    <ng-container *ngIf="dt.isEmpty()">
+      <ng-container *ngTemplateOutlet="dt.emptyMessageTemplate; context: {$implicit: columns}"></ng-container>
+    </ng-container>
+  `
+})
+
+export class TableBodyComponent {
+
+  // tslint:disable-next-line: no-input-rename
+  @Input('bodyColumns') columns;
+
+  @Input() bodyTemplate: TemplateRef<any>;
+
+  constructor(public dt: DataTableComponent) {}
+}
+
+
+
+// Sortable Column Component
+
+@Directive({
+  selector: '[mdSortColumn]',
+  host: {
+    class: 'md-data-table__sortable-column',
+  }
+})
+export class SortColumnDirective implements OnInit {
+
+  @Input('mdSortColumn') field: string;
+
+
+  constructor(public dt: DataTableComponent) { }
+
+  ngOnInit() { }
+
+  @HostListener('click', ['$event'])
+  onClick(event: MouseEvent) {
+
+    this.dt.sort({
+      originalEvent: event,
+      field: this.field
+    });
+  }
+}
+
+
+// Resizable Column
+
+
+@Directive({
```

Comment:
Should we break these directives(`md-sort-column`, `md-resizable-column`, etc.) out into separate files? This one is getting long.

Reply:
I agree, its become lengthy, I'll break them into separate files